include/intrusive_graph.h

```diff
@@ -151,6 +151,13 @@ namespace detail {
 			if(it == deps.end()) return std::nullopt;
 			return it;
 		}
+
+		template <typename Dep>
+		std::optional<typename gch::small_vector<Dep>::const_iterator> maybe_get_dep(const gch::small_vector<Dep>& deps, const T* node) const {
```

Comment:
That's essentially a static method, so there should be no need for const/non-const versions:

```c++
		template <typename Dep>
		static auto maybe_get_dep(gch::small_vector<Dep>& deps, T* node) {
			auto it = std::find_if(deps.begin(), deps.end(), [&](auto d) { return d.node == node; });
			if(it == deps.end()) return std::nullopt; // not sure if this typechecks
			return std::optional{it};
		}
```